### PR TITLE
Feature/add calendar header

### PR DESCRIPTION
**Pull request by kayernyc**
Created a header on the calendar that updates to represent the correct month.
Screen Shot 2020-09-26 at 6 19 28 PM

Used the delegate pattern to inject a handler into the Continuous Scroll Controller.

### CODE REVIEW

File: src/views/virtualDom/ContinuousScrollHandler.ts [hunk 1]
```diff
@@ -207,6 +208,23 @@ export default class ContinuousScrollHandler {
   }
 
   // API
+  headElement(frameElement: HTMLElement = this.frameElement, adoElementDictionary: AdoElementDictionary = this.adoElementDictionary): AtomicDateObject | WeekDateObject {
+    // Each element adds its height until it becomes possitive.
```
Comment (deadnancy) [inline src/views/virtualDom/ContinuousScrollHandler.ts:212]: possitive -> positive.

File: src/views/virtualDom/ContinuousScrollHandler.ts [hunk 2]
```diff
@@ -207,6 +208,23 @@ export default class ContinuousScrollHandler {
...
+    for (const element of elements) {
+      const htmlEL = element as HTMLElement;
+      const key = element.getAttribute(DATA_TAG_STRING);
+      if (offset > -(htmlEL.offsetHeight * 0.5)) {
+
+        return adoElementDictionary[key];
+      }
+
+      offset += htmlEL.offsetHeight;
+    }
```
Comment (deadnancy) [inline src/views/virtualDom/ContinuousScrollHandler.ts:226]: Confused by this. What is it meant to do? Right now if I shorten things to
```
for (const element of elements) {
    const key = element.getAttribute(DATA_TAG_STRING);
    return adoElementDictionary[key];
  }
```
I see no difference. Which makes sense as offset doesn't seem to be applied to anything...?

**Review by deadnancy — APPROVED** [review]
Mostly picky stuff, but nothing I'd call a blocker.

File: src/models/WeekDateObject.ts [hunk 3]
```diff
@@ -21,48 +21,54 @@ export default class WeekDateObject implements AtomicDateObject {
...
-  }
-
-  createInnerHTML = (config: { week: AtomicDateObject[], splitWeek?: boolean, month?: number, today?: Date }): string => {
+  private dateTypeFormat: DateTypeFormat;
+  private today: Date;
+
+  static datesAreSameDay = (candidateDay: Date, today: Date): boolean =>
+    candidateDay.getFullYear() === today.getFullYear() &&
+    candidateDay.getMonth() === today.getMonth() &&
+    candidateDay.getDate() === today.getDate();
```
[the remark this answers is not in the document]
Comment (kayernyc) [reply src/models/WeekDateObject.ts:30]: :)

File: src/views/CalendarView.ts [hunk 4]
```diff
@@ -7,18 +7,29 @@ import DatePickerFactory from '../models/datePickerFactory/DatePickerFactory';
...
+  constructor(model: DatePickerFactory, viewConfiguration: ViewConfiguration) {
     // when min/max is implemented, looping will be possible
     super(model, true, false);
     this.frameElementClassName = 'date-picker-list';
+    this.viewHeader = new ViewHeader();
+  }
+
+  updateIntersectedAdo(ado: AtomicDateObject | WeekDateObject): void {
+    const { date } = ado;
+    this.viewHeader.updateMainText(date.toLocaleString(['en-US'], { month: 'long', year: 'numeric' }));
```
[the remark this answers is not in the document]
Comment (kayernyc) [reply src/views/CalendarView.ts:27]: my prettier is broken >:(

File: src/views/virtualDom/ContinuousScrollHandler.ts (hunk 1 above)
Comment (kayernyc) [reply src/views/virtualDom/ContinuousScrollHandler.ts:212]: sorry for my terrible spelling

File: src/views/virtualDom/ContinuousScrollHandler.ts (hunk 2 above)
Comment (kayernyc) [reply src/views/virtualDom/ContinuousScrollHandler.ts:226]: the offset is how far from the top of the frame element we are. I'm checking for the intersection of the current element with the div that holds the div frame.

Screen Shot 2020-09-28 at 12 20 09 PM

Offset isn't applied to something. It is the distance we have traversed inside the frameElement, which lets us know if the current element is in the visible window or not.

File: src/views/CalendarView.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (deadnancy) [reply src/views/CalendarView.ts:27]: 😢

**Pull request merged**